186090782 Add s3 migration guidance (#542)
alphagov/paas-tech-docs: source/documentation/deploying_services/s3.md

}
```

In the IAM bucket policy, replace:

* the `AWS` value with the `Arn` value from the output of your `aws sts get-caller-identity` command
* the `arn:aws:s3:::` values with the `Arn` value of the bucket you created in your AWS account. You must not paste over the `/*` characters of the first item in the `Resource` list

Your IAM bucket policy should look like this:

```
{
   "Version": "2012-10-17",
   "Statement": [
       {
           "Sid": "PaaS permissions",
           "Effect": "Allow",
           "Principal": {
               "AWS": "<USER_ARN>"
           },
           "Action": [
               "s3:ListBucket",
               "s3:GetObject",
               "s3:PutObject"
           ],
           "Resource": [
               "arn:aws:s3:::<NAME_OF_TENANT_BUCKET>/*",
               "arn:aws:s3:::<NAME_OF_TENANT_BUCKET>"
           ]
       }
   ]
}
```

You can [read more about attaching the IAM bucket policy to the appropriate bucket in the tenant's account](https://docs.aws.amazon.com/AmazonS3/latest/userguide/add-bucket-policy.html).

### Synchronise the S3 buckets

Run this command to synchronise the PaaS S3 bucket and the tenant’s S3 bucket:

```

> aws s3 sync s3://<PAAS_BUCKET_NAME> s3://<TENANT_BUCKET_NAME> --source-region eu-west-1 --region <TENANT-BUCKET-REGION> --profile <NAME_OF_PROFILE_USED_IN_CREDENTIAL_FILE>

Reviewer: ```suggestion
aws s3 sync s3://<PAAS_BUCKET_NAME> s3://<TENANT_BUCKET_NAME> --source-region eu-west-1 --region <TENANT_BUCKET_REGION> --profile <NAME_OF_PROFILE_USED_IN_CREDENTIAL_FILE>
```

```

where `SERVICE_NAME` is a unique descriptive name for this S3 bucket.

## Migrate objects from a PaaS S3 bucket to a S3 bucket in a tenant’s AWS account

You can migrate objects from S3 buckets in PaaS AWS accounts to S3 buckets in other AWS accounts.

You must have an S3 bucket in your AWS account before you begin the migration process. If you do not have an S3 bucket in your AWS account, create one.

The migration process has two main steps:

1. Create an Identity and Access Management (IAM) bucket policy that allows you to copy objects from a PaaS S3 bucket to your S3 bucket.
2. Synchronise the S3 buckets

### Create an IAM bucket policy that allows S3 bucket migration

> To start this migration, you need to apply an IAM bucket policy to your AWS account. This IAM policy will let the S3 broker copy objects from the PaaS S3 bucket to your S3 bucket. 

Reviewer: ```suggestion
To start this migration, you need to apply an IAM bucket policy to your AWS account. This IAM policy will let objects move from the PaaS S3 bucket to your S3 bucket. 
```

{
   "Version": "2012-10-17",
   "Statement": [
       {
           "Sid": "PaaS permissions",
           "Effect": "Allow",
           "Principal": {
               "AWS": "<USER_ARN>"
           },
           "Action": [
               "s3:ListBucket",
               "s3:GetObject",
               "s3:PutObject"
           ],
           "Resource": [
               "arn:aws:s3:::<NAME_OF_TENANT_BUCKET>/*",
               "arn:aws:s3:::<NAME_OF_TENANT_BUCKET>"
           ]
       }
   ]
}
```

You can [read more about attaching the IAM bucket policy to the appropriate bucket in the tenant's account](https://docs.aws.amazon.com/AmazonS3/latest/userguide/add-bucket-policy.html).

### Synchronise the S3 buckets

Run this command to synchronise the PaaS S3 bucket and the tenant’s S3 bucket:

```
aws s3 sync s3://<PAAS_BUCKET_NAME> s3://<TENANT_BUCKET_NAME> --source-region eu-west-1 --region <TENANT-BUCKET-REGION> --profile <NAME_OF_PROFILE_USED_IN_CREDENTIAL_FILE>
```

Replace:

* `<PAAS_BUCKET_NAME>` with the `bucket_name` value provided in the output of the `cf service-key` command
* `<TENANT_BUCKET_NAME>` with the `bucket_name` of the bucket that you created in your AWS account

> * `<TENANT_BUCKET_REGION>` with the `aws region` in which the bucket that you created in your AWS account resides

Reviewer: ```suggestion
* `<TENANT_BUCKET_REGION>` with the [AWS region](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.RegionsAndAvailabilityZones.html#Concepts.RegionsAndAvailabilityZones.Regions) of the bucket that you created in your AWS account
```

Does this work? I've linked out to the AWS docs but I'm not sure if it's the right place to point users. Is there somewhere in the PaaS docs we could link to that explains how to find out the AWS region?
